src/main.ts

```diff
@@ -117,6 +117,24 @@ if (require('electron-squirrel-startup')) {
 
 initializeLogger()
 
+// Used to convert `.json` files into the appropriate file extension for the Generator
+async function migrateJsonGenerator() {
+  const files = (
+    await readdir(GENERATORS_PATH, { withFileTypes: true })
+  ).filter((f) => f.isFile() && path.extname(f.name) === '.json')
+
+  await Promise.all(
+    files.map(async (f) => {
+      const oldPath = path.join(GENERATORS_PATH, f.name)
+      const newPath = path.join(
+        GENERATORS_PATH,
+        path.parse(f.name).name + '.k6g'
+      )
+      await rename(oldPath, newPath)
```

Comment:
You technically don't need to use async/await here.

```suggestion
    files.map((f) => {
      const oldPath = path.join(GENERATORS_PATH, f.name)
      const newPath = path.join(
        GENERATORS_PATH,
        path.parse(f.name).name + '.k6g'
      )
      return rename(oldPath, newPath)
```

Maybe I'm just old fashioned. 😅

Reply:
fair feedback, I actually wondered why I couldn't just await things here but If we make use of `Promise.all` then I guess returning the promise is fine.

It does seem confusing to see a normal function returning a normal value (promise in this case), it's less explicit than having pure async but I might just not be used to the javascript way 😄